action.yml

```diff
@@ -91,8 +91,8 @@ inputs:
     default: false
   aosp-gcc:
     description: 'Use gcc from aosp project.'
-    required: true
-    default: true
+    required: false
```

Comment:
What if neither third-party gcc nor aosp gcc are enabled?

Reply:
then no gcc will be downloaded.

Reply:
however, this will also be useful if users want to use custom clang, as most of custom clang will no longer require a separate gcc.

Reply:
> then no gcc will be downloaded.
I think when both options are not enabled, an error message should be thrown to indicate that at least one option is enabled.

Reply:
But custom clang language also handles binutils replacements like ld.lld, llvm-objcopy, etc. How about using native binutils?

Reply:
> But custom clang also needs to handle binutils replacements like ld.lld, llvm-objcopy, etc. How about using native binutils?

as long as i compile kernel using custom clang, gnu binutils always included in the most of custom clang. only aosp clang who need separate gcc.

Reply:
So can actions automatically recognize the binutils that comes with the third-party clang?

Reply:
> > then no gcc will be downloaded.
> 
> I think when both options are not enabled, an error message should be thrown to indicate that at least one option is enabled.

how about just throwing a warning message? like i said before, downloading gcc is just waste of time if users using custom clang. let users enable it if they need it.

Reply:
So I have a question: can actions automatically recognize the binutils that comes with the third-party clang?

Reply:
> So I have a question: can actions automatically recognize the binutils that comes with the third-party clang?

wait i'll add little statement for it

Reply:
> So I have a question: can actions automatically recognize the binutils that comes with the third-party clang?

check it, it will automatically set NEED_GCC variable to download AOSP GCC when there's no binutils in custom clang.

Reply:
> > > then no gcc will be downloaded.
> > 
> > I think when both options are not enabled, an error message should be thrown to indicate that at least one option is enabled.
> 
> how about just throwing a warning message? like i said before, downloading gcc is just waste of time if users using custom clang. let users enable it if they need it.

Sounds good.